pennylane/workflow/construct_batch.py

```diff
@@ -81,6 +83,110 @@ def _get_full_transform_program(
     return program + qnode.device.preprocess_transforms(config)
 
 
+def _level_sanity_check(
```

Comment:
Can we change the name of this?
```suggestion
def _validate_level(
```